src/pages/Apps/New/:provider/Provider.github.js

```diff
@@ -110,43 +112,89 @@ export default class GithubRepos extends PureComponent {
       width: 1000,
       onClose: async () => {
         const { accounts, selectedAccount } = await this.getInstallations();
-        this.updateState({ accounts, selectedAccount, loading: true });
-        this.getRepositories(selectedAccount);
+        this.updateState({ accounts, selectedAccount });
+        this.getRepositories();
       },
     });
   };
 
-  getRepositories = async (account) => {
-    if (!account) {
+  getRepositories = async () => {
+    // set loading to true to show a spinner until api requests are done
+    this.updateState({ loading: true });
+
+    // get the selected account
+    const { selectedAccount } = this.state;
+
+    if (!selectedAccount) {
       return this.updateState({ loading: false });
     }
 
-    const installationId = account.installationId;
     const api = this.props.github;
-    const res = await api.repositories({ installationId });
+
+    // get current pageIndex and items per page
+    const { repositories, pageQueryParams } = this.state;
+
+    // api params
+    const defaultParams = {
+      page: 1,
+      per_page: 100,
+    };
+    const params =
+      Object.keys(pageQueryParams).length > 0 ? pageQueryParams : defaultParams;
+
+    // fetch the repositories
+    const res = await api.repositories({
+      installationId: selectedAccount.installationId,
+      params,
+    });
+
+    // adds the current page to the repositories
+    const page = res.repositories;
+    repositories.push(...page);
+
+    // check if there is a next page with more repositories
+    const hasNextPage =
+      res.total_count > pageQueryParams.page * pageQueryParams.per_page;
+
+    // gather the params for the next page (if there is one) into an object for consecutive calls through `load more` button
+    const nextPageParams = hasNextPage
+      ? {
+          ...pageQueryParams,
+          page: pageQueryParams.page + 1,
+        }
+      : defaultParams;
 
     this.updateState({
-      selectedAccount: account,
-      repositories: res.repositories,
+      repositories: repositories,
       loading: false,
+      hasNextPage,
+      pageQueryParams: nextPageParams,
     });
   };
 
-  onAccountChange = (login) => {
-    const selectedAccount = this.state.accounts.filter(
-      (a) => a.login === login
-    )[0];
+  onAccountChange = async (login) => {
+    const selectedAccount = this.state.accounts.find((a) => a.login === login);
+    const accounts = this.state.accounts.map((a) => ({
+      ...a,
+      selected: a.login === login,
+    }));
```

Comment:
The `map` function will modify the state object. We should mutating state objects. 

```
JSON.parse(JSON.stringify(this.state.accounts))
```

will create a deep clone of the object and then we can mutate. Perhaps we can even create a new utility function under `src/utils/helpers/array.js` since this seems be something that could be useful in the future. 

The `JSON.parse` and `stringify` method will work as long as the object is serializable (so no functions in the object etc...)

Reply:
good point. what about cloning the object through spreading its properties into a new one:
```
const accounts = [...this.state.accounts].map((a) => ({
  ...a,
  selected: a.login === login,
}));
```

Reply:
This is not exactly a deep clone as if `a` contains nested objects they will be shallow cloned. But for this use case it solves the problem because we're not mutating anything else. So go for it :) Nice suggestion!